ramls/anonymize-loans-request.json

```diff
@@ -0,0 +1,15 @@
+{
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "description": "Request schema for anonymize loans",
+  "type": "object",
+  "properties": {
+    "loanIds": {
```

Comment:
Does it make sense to restrict number of loanId to be processed?

Reply:
I'm not sure, the list is not expected to be very large. It's going to be loans created in one session.

Reply:
https://stackoverflow.com/questions/1009706/postgresql-max-number-of-parameters-in-in-clause
It's stated there that the number of params in `where ... in` clause can be quite large to satisfy our needs.

Reply:
What happens if that is exceeded, will the client receive a server error?

Reply:
A preliminary number of loans to be anonymized in one session will be about 1000 - a number from the PO.

Reply:
If the number is greater than 1000,        it will respond with 422  
```
{
    "errors": [
        {
            "message": "size must be between 0 and 1000",
            "type": "1",
            "code": "-1",
            "parameters": [
                {
                    "key": "loanIds",
                    "value": "[list of 1000 ids]"
                }
            ]
        }
    ]
}
```

Reply:
According to the new details from the PO, the number of items per sesison can be as high as 2,636. @marcjohnson-kint @kostyantyn-kh Do you think it makes sence to restrict number of loanId to be processed to 3000? Alternatively, we can remove the restriction as the number can be quite high.

Reply:
What do you mean by items per session?

What is the limit in the underlying implementation?

Could the client, mod-circulation, submit batches of loans to anonymize?

Reply:
The limit of the implementation is base upon how many Ids we can put into the `where...in` clause and I think that number can be quite large.  I also  think it's best to remove the size restriction here and the clients will submit batches of IDs if needed.

Reply:
Does the size limit defined actually change any behaviour, e.g. if a client submits 1001 loan IDs with the current implementation, what happens?

Reply:
It will be processed successfully. The only limit is the database's ability to handle many items in where...in clause. More info: https://stackoverflow.com/questions/1009706/postgresql-max-number-of-parameters-in-in-clause
One of the concerns for adding this restriction was related to the table row locking. We may experience issues if the table is locked for too long because of update with many items in the where...in clause. 
I still suggest we  remove the restriction here and, if needed, the clients will batch their data.